### PR TITLE
Warn about --follow-untyped-imports

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: docs/source/config_file.rst [hunk 1]
```diff
@@ -320,12 +320,15 @@ section of the command line docs.
...
+
+    Note that analyzing all unannotated modules might result in issues
+    when analyzing code not designed to be type checked and may significantly
+    increase how long mypy takes to run.
 
     If this option is used in a per-module section, the module name should
     match the name of the *imported* module, not the module containing the
-    import statement. Note that scanning all unannotated modules might
-    significantly increase the runtime of your mypy calls.
+    import statement
```
Comment (JelleZijlstra) [inline docs/source/config_file.rst:331]:
```suggestion
    import statement.
```

**Comment by cdce8p** [comment]
Thanks for opening this @hauntsaninja! Does it make sense to add the warning to the `Missing library stubs or py.typed marker` section as well?
https://github.com/python/mypy/blob/71ec4a62f08df4fa28f6b1a9f5bc45c272eaa49f/docs/source/running_mypy.rst?plain=1#L324-L328

**Comment by hauntsaninja** [comment]
Sure, added it there as well

File: docs/source/running_mypy.rst [hunk 2]
```diff
@@ -277,6 +277,25 @@ If you are getting this error, try to obtain type hints for the library you're u
     to the library -- see our documentation on creating
     :ref:`PEP 561 compliant packages <installed-packages>`.
 
+4.  Force mypy to analyze the library as best as it can (as if the library provided
+    a py.typed file), despite it likely missing any type annotations. In general,
```
Comment (cdce8p) [inline docs/source/running_mypy.rst:281]:
```suggestion
    a ``py.typed`` file), despite it likely missing any type annotations. In general,
```

File: docs/source/command_line.rst [hunk 3]
```diff
@@ -168,7 +168,11 @@ imports.
 
 .. option:: --follow-untyped-imports
 
-    This flag makes mypy analyze imports without stubs or a py.typed marker.
+    This flag makes mypy analyze imports even if missing a py.typed marker or stubs.
+
+    Note that analyzing all unannotated modules might result in issues
+    when analyzing code not designed to be type checked and may significantly
+    increase how long mypy takes to run.
```
Comment (cdce8p) [inline docs/source/command_line.rst:175]:
```suggestion
    .. warning::

        Note that analyzing all unannotated modules might result in issues
        when analyzing code not designed to be type checked and may significantly
        increase how long mypy takes to run.
```

File: docs/source/config_file.rst [hunk 4]
```diff
@@ -320,12 +320,15 @@ section of the command line docs.
...
+
+    Note that analyzing all unannotated modules might result in issues
+    when analyzing code not designed to be type checked and may significantly
+    increase how long mypy takes to run.
 
     If this option is used in a per-module section, the module name should
     match the name of the *imported* module, not the module containing the
-    import statement. Note that scanning all unannotated modules might
-    significantly increase the runtime of your mypy calls.
+    import statement.
```
Comment (cdce8p) [inline docs/source/config_file.rst:331]:
```suggestion
    import statement.

    .. warning::

        Note that analyzing all unannotated modules might result in issues
        when analyzing code not designed to be type checked and may significantly
        increase how long mypy takes to run.
```

File: docs/source/config_file.rst [hunk 5]
```diff
@@ -320,12 +320,15 @@ section of the command line docs.
     :type: boolean
     :default: False
 
-    Typechecks imports from modules that do not have stubs or a py.typed marker.
+    Makes mypy analyze imports even if missing a py.typed marker or stubs.
+
+    Note that analyzing all unannotated modules might result in issues
+    when analyzing code not designed to be type checked and may significantly
+    increase how long mypy takes to run.
```
Comment (cdce8p) [inline docs/source/config_file.rst:327]:
```suggestion
```

File: docs/source/command_line.rst [hunk 6]
```diff
@@ -168,7 +168,11 @@ imports.
 
 .. option:: --follow-untyped-imports
 
-    This flag makes mypy analyze imports without stubs or a py.typed marker.
+    This flag makes mypy analyze imports even if missing a py.typed marker or stubs.
```
Comment (JukkaL) [inline docs/source/command_line.rst:171]: Might be good to explicitly mention that this is about installed packages, including those installed with pip? For example, something along these lines:
> This flag makes mypy analyze imported installed packages (including third-party packages installed using pip), even if they don't explicitly declare support for type checking by providing a py.typed marker or stubs.

Maybe also add a link from "py.typed marker" to relevant external documentation?

File: docs/source/running_mypy.rst [hunk 7]
```diff
@@ -277,6 +277,25 @@ If you are getting this error, try to obtain type hints for the library you're u
...
+    This option can be specified on a per-module basis as well::
+
+        # mypy.ini
+        [mypy-untyped_package.*]
+        follow_untyped_imports = True
+
+        # pyproject.toml
+        [[tool.mypy.overrides]]
+        module = ["untyped_package.*"]
+        follow_untyped_imports = true
```
Comment (cdce8p) [inline docs/source/running_mypy.rst:297]: We could use `sphinx_inline_tabs` for that. Opened #18262 for it.

**Pull request merged**